packages/web3-types/src/eth_types.ts

```diff
@@ -367,8 +367,10 @@ export interface TransactionCall extends Transaction {
 	to: Address;
 }
 
-export interface TransactionWithLocalWalletIndex extends TransactionBase {
+export interface TransactionWithLocalWalletIndex extends Omit<TransactionBase, 'to'> {
 	from?: Numbers;
+	// eslint-disable-next-line @typescript-eslint/ban-types
+	to?: Address | Numbers | null;
```

Comment:
`to` being `Address | null` is already handled by the type [TransactionBase](https://github.com/web3/web3.js/blob/c4e255d277a6b57a92e0325eefa79852ba7dd864/packages/web3-types/src/eth_types.ts#L337), so it should just be listed as `Numbers` here. The problem is `from` could be the local wallet index while `to` is `Address | null`, so I see why you did this, but it doesn't work for the inverse where `to` is an index but `from` is `Address | null`

I'm not sure how else to this besides introducing the following types: `TransactionWithFromLocalWalletIndex`, `TransactionWithToLocalWalletIndex`, and `TransactionWithFromAndToLocalWalletIndex`, so definitely open to suggestions. I've opened #5748 to implement this (doesn't update the `CHANGELOG`s, so that will need to be done if you decided to merge it)